File: engine/engine-model/src/main/scala/tools/packet/remove/RemoveSummon.scala
package tools.packet.remove

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class RemoveSummon(private var _ownerId: Int, private var _objectId: Int, private var _animated: Boolean) extends PacketInput {
  def ownerId: Int = _ownerId

  def objectId: Int = _objectId

  def animated: Boolean = _animated

  override def opcode(): SendOpcode = SendOpcode.REMOVE_SPECIAL_MAPOBJECT
}

File: engine/engine-model/src/main/java/tools/packet/remove/RemoveSummon.java
package tools.packet.remove;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record RemoveSummon(Integer ownerId, Integer objectId, Boolean animated) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.REMOVE_SPECIAL_MAP_OBJECT;
   }
}